base/sparse/cholmod.jl

```diff
@@ -231,836 +181,892 @@ else
 
-function solve{Tv<:CHMVTypes,Ti<:CHMITypes}(L::CholmodFactor{Tv,Ti},
-                                            B::SparseMatrixCSC{Tv,Ti},typ::Integer)
-    sparse!(solve(L,CholmodSparse(B),typ))
+free!(A::Dense) = free_dense(A.p)
+free!(A::Sparse) = free_sparse(A.p)
+free!(F::Factor) = free_factor(F.p)
+
+eltype{T<:VTypes}(A::Dense{T}) = T
+eltype{T<:VTypes}(A::Factor{T}) = T
+eltype{T<:VTypes}(A::Sparse{T}) = T
+
+function show(io::IO, F::Factor)
+    s = unsafe_load(F.p)
+    println(io, typeof(F))
+    @printf(io, "type: %12s\n", bool(s.is_ll) ? "LLt" : "LDLt")
+    @printf(io, "method: %10s\n", bool(s.is_super) ? "supernodal" : "simplicial")
+    @printf(io, "maxnnz: %10d\n", int(s.nzmax))
+    @printf(io, "nnz: %13d\n", nnz(Sparse(F)))
 end
-function solve{Tv<:CHMVTypes,Ti<:CHMITypes}(L::CholmodFactor{Tv,Ti},B::SparseMatrixCSC{Tv,Ti})
-    sparse!(solve(L,CholmodSparse(B),CHOLMOD_A))
+
+isvalid(A::Dense) = check_dense(A)
+isvalid(A::Sparse) = check_sparse(A)
+isvalid(A::Factor) = check_factor(A)
+
+copy(A::Dense) = copy_dense(A)
+copy(A::Sparse) = copy_sparse(A)
+copy(A::Factor) = copy_factor(A)
+
+function size(A::Union(Dense,Sparse))
+    s = unsafe_load(A.p)
+    return (int(s.nrow), int(s.ncol))
 end
-function solve{Tv<:CHMVTypes,Ti<:CHMITypes}(L::CholmodFactor{Tv,Ti},B::CholmodSparse{Tv,Ti})
-    solve(L,B,CHOLMOD_A)
+function size(F::Factor, i::Integer)
+    i < 1 && error("arraysize: dimension out of range")
+    s = unsafe_load(F.p)
+    if i <= 2
+        return int(s.n)
+    end
+    return 1
 end
-function (\){Tv<:CHMVTypes,Ti<:CHMITypes}(L::CholmodFactor{Tv,Ti},B::CholmodSparse{Tv,Ti})
-    solve(L,B,CHOLMOD_A)
+
+function getindex(A::Dense, i::Integer)
+    s = unsafe_load(A.p)
+    0 < i <= s.nrow*s.ncol || throw(BoundsError())
+    unsafe_load(s.x, i)
 end
-function solve{T<:CHMVTypes}(L::CholmodFactor{T},B::Vector{T},typ::Integer)
-    solve(L,CholmodDense!(B),typ).mat[:]
+function getindex(A::Dense, i::Integer, j::Integer)
+    s = unsafe_load(A.p)
+    0 < i <= s.nrow || throw(BoundsError())
+    0 < j <= s.ncol || throw(BoundsError())
+    unsafe_load(s.x, i + (j - 1)*s.d)
 end
-function solve{T<:CHMVTypes}(L::CholmodFactor{T},B::Matrix{T},typ::Integer)
-    solve(L,CholmodDense!(B),typ).mat
+
+getindex(A::Sparse, i::Integer) = getindex(A, ind2sub(size(A),i)...)
+function getindex{T}(A::Sparse{T}, i0::Integer, i1::Integer)
+    s = unsafe_load(A.p)
+    !(1 <= i0 <= s.nrow && 1 <= i1 <= s.ncol) && throw(BoundsError())
+    s.stype < 0 && i0 < i1 && return conj(A[i1,i0])
+    s.stype > 0 && i0 > i1 && return conj(A[i1,i0])
+
+    r1 = int(unsafe_load(s.p, i1) + 1)
+    r2 = int(unsafe_load(s.p, i1 + 1))
+    (r1 > r2) && return zero(T)
+    r1 = int(searchsortedfirst(pointer_to_array(s.i, (s.nzmax,), false), i0 - 1, r1, r2, Base.Order.Forward))
+    ((r1 > r2) || (unsafe_load(s.i, r1) + 1 != i0)) ? zero(T) : unsafe_load(s.x, r1)
```

Comment:
You are probably right. I can probably reach all of them by testing the show method on `Sparse`.